kernel/src/hil/flash.rs

```diff
@@ -12,14 +16,14 @@ pub trait Flash {
     /// when operations complete.
     fn set_client(&self, client: &'static Client);
 
-    /// Read data
-    fn read(&self, offset: usize, buf: &'static mut [u8]);
+    /// Read a page of flash.
+    fn read_page(&self, page_number: usize, buf: &'static mut [u8]) -> ReturnCode;
 
-    /// Write data
-    fn write(&self, offset: usize, buf: &'static mut [u8]);
+    /// Write a page of flash.
+    fn write_page(&self, page_number: usize, buf: &'static mut [u8]) -> ReturnCode;
```

Comment:
Right... So ideally, then, the `buf` arguments would be reference to fixed size arrays. Or is there a problem that different flash controllers have different page sizes?

So something like:

```rust
fn write_page(&self, page_number: usize, buf: &'static mut [u8; 4096])
```

would be too restrictive for a general trait?